render/reply/protobuf.go

```diff
@@ -71,6 +73,18 @@ func replyProtobuf(p pb, w http.ResponseWriter, r *http.Request, multiData data.
 				p.writeBody(writer, a.Target, a.DisplayName, function, from, until, step, points)
 			}
 		}
+
+		// fill metrics without points with NaN
+		if d.AppendOutEmptySeries && len(writtenMetrics) < data.AM.Len() && data.CommonStep > 0 {
+			for _, metricName := range data.AM.Series(false) {
```

Comment:
May be simplify to
`if _, done := writtenMetrics[metricName]; !done {`
and avoid of continue block ?